src/sync/channel.rs

```diff
@@ -58,6 +58,7 @@ use crate::sync::WakerSet;
 #[cfg(feature = "unstable")]
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
 pub fn channel<T>(cap: usize) -> (Sender<T>, Receiver<T>) {
+    assert!(cap >= 1);
```

Comment:
Can we change this into a `debug_assert`?
```suggestion
    debug_assert!(cap >= 1);
```

Reply:
We can, the goal here was to pull this assertation:
https://github.com/async-rs/async-std/blob/master/src/sync/channel.rs#L638 forward to the 'offending' function so that it's easier to track the issue.